### PR TITLE
Add table aws_inspector_assessment_target. closes #226

### LINKED ISSUE #226: Add table aws_inspector_assessment_target
**References**
https://pkg.go.dev/github.com/aws/aws-sdk-go/service/inspector

### PR DESCRIPTION
**Pull request by sayan133**
# Integration test logs
<details>
  <summary>Logs</summary>
  
```
No env file present for the current environment:  staging 
 Falling back to .env config
No env file present for the current environment:  staging
customEnv TURBOT_TEST_EXPECTED_TIMEOUT undefined

SETUP: tests/aws_inspector_assessment_target []

PRETEST: tests/aws_inspector_assessment_target

TEST: tests/aws_inspector_assessment_target
Running terraform
aws_inspector_resource_group.inspector_rg: Creating...
aws_inspector_resource_group.inspector_rg: Creation complete after 3s [id=arn:aws:inspector:us-east-1:388460667113:resourcegroup/0-lMf2q9bo]
aws_inspector_assessment_target.named_test_resource: Creating...
aws_inspector_assessment_target.named_test_resource: Creation complete after 2s [id=arn:aws:inspector:us-east-1:388460667113:target/0-1A0YZ4h8]

Warning: Deprecated Resource

The null_data_source was historically used to construct intermediate values to
re-use elsewhere in configuration, the same can now be achieved using locals


Apply complete! Resources: 2 added, 0 changed, 0 destroyed.

Outputs:

account_id = "388460667113"
aws_partition = "aws"
aws_region = "us-east-1"
resource_aka = "arn:aws:inspector:us-east-1:388460667113:target/0-1A0YZ4h8"
resource_group_aka = "arn:aws:inspector:us-east-1:388460667113:resourcegroup/0-lMf2q9bo"
resource_name = "turbottest9791"

Running SQL query: test-get-query.sql
[
  {
    "account_id": "388460667113",
    "arn": "arn:aws:inspector:us-east-1:388460667113:target/0-1A0YZ4h8",
    "name": "turbottest9791",
    "partition": "aws",
    "region": "us-east-1",
    "resource_group_arn": "arn:aws:inspector:us-east-1:388460667113:resourcegroup/0-lMf2q9bo"
  }
]
✔ PASSED

Running SQL query: test-list-query.sql
[
  {
    "arn": "arn:aws:inspector:us-east-1:388460667113:target/0-1A0YZ4h8",
    "name": "turbottest9791",
    "resource_group_arn": "arn:aws:inspector:us-east-1:388460667113:resourcegroup/0-lMf2q9bo"
  }
]
✔ PASSED

Running SQL query: test-notfound-query.sql
null
✔ PASSED

Running SQL query: test-turbot-query.sql
[
  {
    "akas": [
      "arn:aws:inspector:us-east-1:388460667113:target/0-1A0YZ4h8"
    ],
    "title": "turbottest9791"
  }
]
✔ PASSED

POSTTEST: tests/aws_inspector_assessment_target

TEARDOWN: tests/aws_inspector_assessment_target

SUMMARY:

1/1 passed.


```
</details>

# Example query results
<details>
  <summary>Results</summary>
  
```
### Basic info

```sql
select
  name,
  arn,
  resource_group_arn,
  created_at,
  updated_at,
  region
from
  aws_inspector_assessment_target;
```

```
+-----------+------------------------------------------------------------+-------------------------------------------------------------------+---------------------+---------------------+-----------+
| name      | arn                                                        | resource_group_arn                                                | created_at          | updated_at          | region    |
+-----------+------------------------------------------------------------+-------------------------------------------------------------------+---------------------+---------------------+-----------+
| test-stem | arn:aws:inspector:us-east-1:388460667113:target/0-O7bVKxDE | arn:aws:inspector:us-east-1:388460667113:resourcegroup/0-DjVxQgCg | 2021-03-22 08:11:08 | 2021-03-22 08:11:08 | us-east-1 |
+-----------+------------------------------------------------------------+-------------------------------------------------------------------+---------------------+---------------------+-----------+

```

### List Assessment Targets that are created within last 7 days

```sql
select
  name,
  arn,
  resource_group_arn,
  created_at,
  updated_at,
  region
from
  aws_inspector_assessment_target
where
  created_at > (current_date - interval '7' day);
```

```
+-----------+------------------------------------------------------------+-------------------------------------------------------------------+---------------------+---------------------+-----------+
| name      | arn                                                        | resource_group_arn                                                | created_at          | updated_at          | region    |
+-----------+------------------------------------------------------------+-------------------------------------------------------------------+---------------------+---------------------+-----------+
| test-stem | arn:aws:inspector:us-east-1:388460667113:target/0-O7bVKxDE | arn:aws:inspector:us-east-1:388460667113:resourcegroup/0-DjVxQgCg | 2021-03-22 08:11:08 | 2021-03-22 08:11:08 | us-east-1 |
+-----------+------------------------------------------------------------+-------------------------------------------------------------------+---------------------+---------------------+-----------+

```

### List Assessment Targets that are updated after creation

```sql
select
  name,
  arn,
  resource_group_arn,
  created_at,
  updated_at,
  region
from
  aws_inspector_assessment_target
where
  created_at != updated_at;
```

```
+------+-----+--------------------+------------+------------+--------+
| name | arn | resource_group_arn | created_at | updated_at | region |
+------+-----+--------------------+------------+------------+--------+
+------+-----+--------------------+------------+------------+--------+

```
</details>

### CODE REVIEW

File: aws-test/tests/aws_inspector_assessment_target/variables.tf [hunk 1]
```diff
@@ -0,0 +1,83 @@
+variable "resource_name" {
+  type        = string
+  default     = "turbot-test-20200125-create-update"
+  description = "Name of the resource used throughout the test."
+}
+
+variable "aws_profile" {
+  type        = string
+  default     = "integration-tests"
```
Comment (khushboo9024) [inline aws-test/tests/aws_inspector_assessment_target/variables.tf:9]:
```suggestion
  default     = "default"
```

File: aws/table_aws_inspector_assessment_target.go [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+
+//// TABLE DEFINITION
+
+func tableAwsInspectorAssessmentTarget(_ context.Context) *plugin.Table {
+	return &plugin.Table{
+		Name:        "aws_inspector_assessment_target",
+		Description: "AWS Inspector Assessment Target",
+		Get: &plugin.GetConfig{
+			KeyColumns:        plugin.SingleColumn("arn"),
+			ShouldIgnoreError: isNotFoundError([]string{}),
```
Comment (khushboo9024) [inline aws/table_aws_inspector_assessment_target.go:21]: If there is no not found error then please remove this.

File: aws/table_aws_inspector_assessment_target.go [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+				Transform:   transform.FromField("Arn").Transform(arnToAkas),
+			},
+		}),
+	}
+}
+
+//// LIST FUNCTION
+
+func listAssessmentTargets(ctx context.Context, d *plugin.QueryData, _ *plugin.HydrateData) (interface{}, error) {
+	// TODO put me in helper function
```
Comment (khushboo9024) [inline aws/table_aws_inspector_assessment_target.go:80]:
```suggestion
```

File: aws/table_aws_inspector_assessment_target.go [hunk 4]
```diff
@@ -0,0 +1,151 @@
...
+	return nil, err
+}
+
+//// HYDRATE FUNCTIONS
+
+func describeAssessmentTarget(ctx context.Context, d *plugin.QueryData, h *plugin.HydrateData) (interface{}, error) {
+	logger := plugin.Logger(ctx)
+	logger.Trace("describeAssessmentTarget")
+
+	// TODO put me in helper function
```
Comment (khushboo9024) [inline aws/table_aws_inspector_assessment_target.go:116]:
```suggestion
```

File: aws/table_aws_inspector_assessment_target.go [hunk 5]
```diff
@@ -0,0 +1,151 @@
...
+			return !isLast
+		},
+	)
+
+	return nil, err
+}
+
+//// HYDRATE FUNCTIONS
+
+func describeAssessmentTarget(ctx context.Context, d *plugin.QueryData, h *plugin.HydrateData) (interface{}, error) {
```
Comment (khushboo9024) [inline aws/table_aws_inspector_assessment_target.go:112]: function name should be `getInspectorAssessmentTarget`

File: aws/table_aws_inspector_assessment_target.go [hunk 6]
```diff
@@ -0,0 +1,151 @@
...
+				Type:        pb.ColumnType_JSON,
+				Transform:   transform.FromField("Arn").Transform(arnToAkas),
+			},
+		}),
+	}
+}
+
+//// LIST FUNCTION
+
+func listAssessmentTargets(ctx context.Context, d *plugin.QueryData, _ *plugin.HydrateData) (interface{}, error) {
```
Comment (khushboo9024) [inline aws/table_aws_inspector_assessment_target.go:79]: function name should be `listInspectorAssessmentTargets`

**Review by khushboo9024 — COMMENTED** [review]
Please make the suggested changes.

**Review by khushboo9024 — APPROVED** [review]
LGTM

File: aws/table_aws_inspector_assessment_target.go [hunk 7]
```diff
@@ -0,0 +1,148 @@
...
+		AssessmentTargetArns: []*string{aws.String(assessmentTargetArn)},
+	}
+
+	// Get call
+	data, err := svc.DescribeAssessmentTargets(params)
+	if err != nil {
+		logger.Debug("describeAssessmentTarget__", "ERROR", err)
+		return nil, err
+	}
+	if len(data.AssessmentTargets) > 0 {
```
Comment (Subhajit97) [inline aws/table_aws_inspector_assessment_target.go:144]:
```suggestion
	if data.AssessmentTargets != nil && len(data.AssessmentTargets) > 0 {
```

**Review by Subhajit97 — CHANGES_REQUESTED** [review]
@sayan133 There is a small change, otherwise looks good.

**Review by Subhajit97 — APPROVED** [review]
LGTM

**Review by rajmohanty17 — APPROVED** [review]
LGTM

**Review by cbruno10 — APPROVED** [review]

**Pull request merged**